datashader/utils.py

```diff
@@ -469,10 +479,127 @@ def mesh(vertices, simplices):
                                 'consider casting simplices to integers '
                                 'with ".astype(int)"')
 
-    assert len(vertices.columns) > 2 or simplices.values.shape[1] > 3, 'If no vertex weight column is provided, a triangle weight column is required.'
-
+    assert len(vertices.columns) > 2 or simplices.values.shape[1] > 3, \
+        'If no vertex weight column is provided, a triangle weight column is required.'
 
-    if isinstance(vertices, dd.DataFrame) and isinstance(simplices, dd.DataFrame):
+    if has_dask and isinstance(vertices, DaskDataFrame) and isinstance(simplices, DaskDataFrame):
         return _dd_mesh(vertices, simplices)
 
     return _pd_mesh(vertices, simplices)
+
+
+def pyspark_type_to_pandas_type(data_type):
+    """
+    Map a PySpark type to a pandas (numpy) type
+    """
+    import pyspark.sql.types as T
+    if isinstance(data_type, T.ArrayType):
+        return np.object
+    if isinstance(data_type, T.BooleanType):
+        return np.bool
+    if isinstance(data_type, T.ByteType):
+        return np.int8
+    if isinstance(data_type, T.DecimalType):
+        return np.float64
+    if isinstance(data_type, T.DoubleType):
+        return np.float64
+    if isinstance(data_type, T.FloatType):
+        return np.float32
+    if isinstance(data_type, T.IntegerType):
+        return np.int32
+    if isinstance(data_type, T.LongType):
+        return np.int64
+    if isinstance(data_type, T.MapType):
+        return np.object
+    if isinstance(data_type, T.NullType):
+        return np.object
+    if isinstance(data_type, T.ShortType):
+        return np.int16
+    if isinstance(data_type, T.StructType):
+        return np.object
+    if isinstance(data_type, T.StringType):
+        return np.object
+    if isinstance(data_type, T.TimestampType):
+        return np.datetime64
+    raise TypeError("Unrecognized PySpark type %r" % data_type)
```

Comment:
I'd prefer if this were a dictionary plus a line or two of code, not a bunch of ifs.